TensorIO/tensorio/src/main/java/ai/doc/tensorio/TIOModel/TIOModelIO.java

```diff
@@ -1,4 +1,168 @@
 package ai.doc.tensorio.TIOModel;
 
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.Set;
+
+import ai.doc.tensorio.TIOLayerInterface.TIOLayerInterface;
+
+/**
+ * Encapsulates information about the inputs, outputs, and placeholders for a model.
+ */
+
 public class TIOModelIO {
+
+    /**
+     * An I/O list may be indexed by key or by index.
+     */
+
+    public class TIOModelIOList {
+
+        private List<TIOLayerInterface> indexedInterfaces;
+        private Map<String, TIOLayerInterface> namedInterfaces;
+        private Map<String, Integer> nameToIndex;
+
+        /**
+         * Initializes an indexed model list with a list interfaces. You should not
+         * need to create instances of this class yourself.
+         *
+         * If the initializing interfaces parameter is nil, it will be treated as an
+         * empty list.
+         */
+
+        public TIOModelIOList(@Nullable List<TIOLayerInterface> interfaces) {
```

Comment:
Adding info about the parameters would look great on function definition. 

"/**
  * Initializes an indexed model list with a list interfaces. You should not
  * need to create instances of this class yourself.
  *
  * If the initializing interfaces parameter is nil, it will be treated as an
  * empty list.

  * @param interfaces A list of TIOLayerInterface Objects (or something similar)
*/
"

Reply:
oh yeah for sure, i copied these almost verbatim from the objective-c and sometimes i'm lazy about adding everything there